src/nats.h

```diff
@@ -4564,6 +4564,28 @@ natsSubscription_AutoUnsubscribe(natsSubscription *sub, int max);
 NATS_EXTERN natsStatus
 natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs);
 
+/** \brief Gets the subscription id.
+ *
+ * Returns the id of the given subscription.
+ *
+ * @param sub the pointer to the #natsSubscription object.
+ * @param id if not `NULL`, the memory location where to store the subscription id.
+ */
+NATS_EXTERN natsStatus
+natsSubscription_GetID(natsSubscription* sub, int64_t* id);
+
+/** \brief Gets the subject name.
+ *
+ * Returns the subject of the given subscription.
+ *
+ * \warning The string belongs to the subscription and must not be freed. Copy it if needed.
+ *
+ * @param sub the pointer to the #natsSubscription object.
+ * @param subject if not `NULL`, the memory location where to store the subject name.
+ */
+NATS_EXTERN natsStatus
+natsSubscription_GetSubject(natsSubscription* sub, const char* subject);
```

Comment:
If we keep return status and return value through "out" parameter, then you need to have `const char **subject` here.